lib/aws-cleaner.rb

```diff
@@ -30,15 +30,15 @@ def self.client(config)
     # call the Chef API to get the node name of the instance
     def self.get_chef_node_name(instance_id, config)
       chef = client(config)
-      results = chef.search.query(:node, "ec2_instance_id:#{instance_id} OR chef_provisioning_reference_server_id:#{instance_id}")
+      results = chef.search.query(:node, "ec2_instance_id:#{instance_id[2..-1]} OR chef_provisioning_reference_server_id:#{instance_id[2..-1]}")
```

Comment:
I don't have a chef server currently to test this against can you show me the output of this or explain this change?

Reply:
This removes the "i-" part from the instance ID in case you don't have that. I do the same on the sensu side since we don't have that part in our client names.

I tested it on a quick chef install and it seemed to work but don't have a full chef install to test against. I'm ok omitting this change.

Reply:
I'd like to address this separately and will be starting a new job next week so I will have a chef and sensu server to test with.

Reply:
I think that should be removed since it's going to break lookups for the instance. Ohai populates that attribute with the actual instance-id and it will always include `i-`.

Reply:
I have reverted that, I figured since it was the "search" api that was being called it would also search partial strings it seemed to work on my quick test with chef but since I don't use it I'm fine with reverting it.